include/translatable-object.php

```diff
@@ -391,24 +391,30 @@ public function get_objects_with_no_lang( $limit, array $args = array() ) {
 			return array();
 		}
 
-		$sql        = $this->get_objects_with_no_lang_sql( $language_ids, $limit, $args );
-		$object_ids = $this->query_objects_with_no_lang( $sql );
+		$object_ids = $this->query_objects_with_no_lang( $language_ids, $limit, $args );
 
 		return array_values( $this->sanitize_int_ids_list( $object_ids ) );
 	}
 
 	/**
-	 * Returns object IDs without language given a specific SQL query.
+	 * Returns object IDs without language.
 	 * Can be overridden by child classes in case queried object doesn't use
 	 * `wp_cache_set_last_changed()` or another cache system.
 	 *
 	 * @since 3.4
+	 * @since 3.7 Changed all parameters.
 	 *
-	 * @param string $sql A prepared SQL query for object IDs with no language.
+	 * @param int[] $language_ids List of language `term_taxonomy_id`.
+	 * @param int   $limit        Max number of objects to return. `-1` to return all of them.
+	 * @param array $args         The object args.
 	 * @return string[] An array of numeric object IDs.
+	 *
+	 * @phpstan-param array<positive-int> $language_ids
+	 * @phpstan-param -1|positive-int $limit
+	 * @phpstan-param array<empty> $args
 	 */
-	protected function query_objects_with_no_lang( $sql ) {
-		$key          = md5( $sql );
+	protected function query_objects_with_no_lang( array $language_ids, $limit, array $args = array() ) {
+		$key          = md5( maybe_serialize( $language_ids ) . maybe_serialize( $args ) . $limit );
```

Comment:
```suggestion
		$key          = md5( (string) json_encode( $language_ids + $args ) . $limit );
```
To improve perf a bit.

Reply:
I prefer not to merge the 2 different arrays